src/daft-plan/src/logical_plan.rs

```diff
@@ -224,10 +224,10 @@ impl LogicalPlan {
                 Self::Project(Project { projection, resource_request, .. }) => Self::Project(Project::try_new(
                     input.clone(), projection.clone(), resource_request.clone(),
                 ).unwrap()),
-                Self::Filter(Filter { predicate, .. }) => Self::Filter(Filter::new(predicate.clone(), input.clone())),
+                Self::Filter(Filter { predicate, .. }) => Self::Filter(Filter::try_new(predicate.clone(), input.clone()).unwrap()),
```

Comment:
Agreed! I thought about making both changes in this PR, actually, but thought that it would be too much of a driveby.

I think that I would most be in favor of the former, but it also how we decide to handle internal optimization/planning errors: we can either panic or return a nicer error to the user. I think we should panic for now?